proxy/src/connection.rs

```diff
@@ -184,15 +184,15 @@ impl Future for Connecting {
 
     fn poll(&mut self) -> Poll<Self::Item, Self::Error> {
         loop {
-            let new_state = match self {
+            *self = match self {
```

Comment:
👍

Reply:
Incidentally, I really don't like that we have this explicit `Future` implementation. I would prefer it to be rewritten in terms of combinators but I couldn't get that to work after spending some time spent on it. Also, since we're going to implement some "fall back to non-secure if TLS failed" logic, we should probably do that fallback logic first and then see if we can simplify things.

Reply:
Yeah, I think that sounds like the right approach.